### ISSUE TITLE
Cron validation fails with false positive

### ISSUE THREAD

**Issue by manolitooo** [HangfireIO/Hangfire#2100]
Hi,

Cron validation is failing, even though the Cronos library underneath supports the expression. Expressions whose validation fails as a false positive are those which start with "@", and the problem is in this piece of code:

```csharp
        public static CronExpression ParseCronExpression([NotNull] string cronExpression)
        {
            if (cronExpression == null) throw new ArgumentNullException(nameof(cronExpression));

            var parts = cronExpression.Split(new[] { ' ', '\t' }, StringSplitOptions.RemoveEmptyEntries);
            var format = CronFormat.Standard;

            if (parts.Length == 6)
            {
                format |= CronFormat.IncludeSeconds;
            }
            else if (parts.Length != 5)
            {
                throw new CronFormatException(
                    $"Wrong number of parts in the `{cronExpression}` cron expression, you can only use 5 or 6 (with seconds) part-based expressions.");
            }

            return CronExpression.Parse(cronExpression, format);
        }
```

You can try with "@midnight", for example. It is meant to run a job every day at midnight. And it is supported by the Cronos library. I think it could be easily fixed by putting a first "IF" like:

```csharp
            if (!cronExpression.StartsWith("@"))
            {
                if (parts.Length == 6)
                {
                    format |= CronFormat.IncludeSeconds;
                }
                else if (parts.Length != 5)
                {
                    throw new CronFormatException(
                        $"Wrong number of parts in the `{cronExpression}` cron expression, you can only use 5 or 6 (with seconds) part-based expressions.");
                }
            }
```

Or (not sure, it depends on the influence of "format" parameter later):

```csharp
            if (parts.Length == 6 || cronExpression.StartsWith("@"))
            {
                [...]
```

Regards.

**Comment by MuhamedAbdalla** [comment]
@odinserj 
I'm willing to work on this issue, Can you assign it to me please?

**Comment by odinserj** [comment]
Sure, just did it, thank you for your intention! I will be happy to get a PR for this.

**Comment by MuhamedAbdalla** [comment]
@odinserj 
Sorry for being late, I just saw the message. Thanks for your consideration Sir <3 
This is the PR for it. I'll appreciate if you can take a look.
https://github.com/HangfireIO/Hangfire/pull/2323

**Comment by MuhamedAbdalla** [comment]
@odinserj 
Sorry for being late. Just saw the notification. Thanks a lot for your consideration Sir <3 
This is the PR, I'd appreciate if you can take a look on it.

https://github.com/HangfireIO/Hangfire/pull/2324

**Comment by RahulYA2020** [comment]
Hi @odinserj ,
Is this issue still open?

**Comment by odinserj** [comment]
Hi, actually fixed in 1.8.7, so should be working now - https://github.com/HangfireIO/Hangfire/releases.
